MonoGame.Framework/Input/GamePadState.cs

```diff
@@ -82,7 +96,7 @@ public GamePadState(GamePadThumbSticks thumbSticks, GamePadTriggers triggers, Ga
         /// <param name="leftTrigger">Left trigger value. This value is clamped between 0.0 and 1.0.</param>
         /// <param name="rightTrigger">Right trigger value. This value is clamped between 0.0 and 1.0.</param>
         /// <param name="buttons"> Array or parameter list of Buttons to initialize as pressed.</param>
-        public GamePadState(Vector2 leftThumbStick, Vector2 rightThumbStick, float leftTrigger, float rightTrigger, Buttons buttons)
+        public GamePadState(Vector2 leftThumbStick, Vector2 rightThumbStick, float leftTrigger, float rightTrigger, params Buttons[] buttons)

```

Comment:
Please remove the `params` here. While it is mentioned like this in #6027 the XNA constructor did not have a `params` argument (just an array) and we want to prevent the implicit allocation.

Reply:
Updated the pull request. Thanks for the feedback!

Reply:
The parameter needs to be `params Buttons[] buttons`. 
In XNA you could create a GamePadState() with any number of buttons parameters. 
e.g. `new GamePadState(leftTS, rightTS, leftT, rightT, Buttons.DPadDown, DPadRight.Buttons)`